### ISSUE TITLE
filenames in compiler errors to not contain enough path information to disambiguate , confusing visual studio

### ISSUE THREAD

**Issue by leecow** [dotnet/project-system#5588]
@Spongman commented on [Mon Jun 04 2018](https://github.com/dotnet/core/issues/1656)

when i build one of the dotnet core projects in my solution, i get the following error:

```
1>------ Rebuild All started: Project: MyProject, Configuration: Debug Any CPU ------
1>Program.cs(58,4,58,30): error CS0234: The type or namespace name 'ServicePointManager' does not exist in the namespace 'System' (are you missing an assembly reference?)
1>Done building project "MyProject.csproj" -- FAILED.
========== Rebuild All: 0 succeeded, 1 failed, 0 skipped ==========
```

when i double-click on that error in the Visual Studio output window it opens the _wrong_ `Program.cs` file. this is no doubt due to the fact that the full solution-relative path to the file containing the error isn't included.

when building with the desktop csc.exe the full (ie c:\\-relative) path is displayed so there is no ambiguity.

**Comment by davkean** [comment]
Moved this over to us, @rainersigwald @nguerrera I'm making this bug about the project-system changes to make this occur. If you want to track the MSBuild changes/CLI/SDK making a change here, you'll need a different bug. We won't block on that decision.

**Comment by davkean** [comment]
I've made the change to VS here: https://devdiv.visualstudio.com/DevDiv/_git/CPS/pullrequest/211126?_a=overview (Microsofties only).

**Comment by davkean** [comment]
This has been fixed and will be available in the first preview of 16.5.

**Comment by davkean** [comment]
Note, the fix I made for this [introduced a regression](https://developercommunity.visualstudio.com/content/problem/1000120/165-introduced-a-new-msbuild-property-generatefull.html). We're fixing this and making a different fix, but we want MSBuild to ultimately flip this default: https://github.com/microsoft/msbuild/issues/5358.
